### ISSUE TITLE
Figure out a better way to generate a file name

### ISSUE THREAD

**Issue by zpeters** [zpeters/stashbox#7]
Currently it is taking a hash of the path.  It would be nice to pull the webpage title, similar to what a web browser would do when you save a file

**Comment by lucasturci** [comment]
Hi! I think the web browser extracts what is in the <title> tag of the HTML. Is that you want? I can do this.

**Comment by zpeters** [comment]
I think that would work perfectly

**Comment by lucasturci** [comment]
Great! Can you assign me the issue?

**Comment by zpeters** [comment]
assigned, thank you!

**Comment by AceroM** [comment]
@lucasturci  you could use regexp to find the title or parse it with the [golang.org/x/net/html](https://pkg.go.dev/golang.org/x/net/html#Parse) package.

so something like this could work
```
    var title string
	htmlBody, err := ioutil.ReadAll(resp.Body)
	titleRegex := `.*?<title>(.*?)</title>.*`
	re := regexp.MustCompile(titleRegex)
	matches := re.FindSubmatch(htmlBody)
	if len(matches) > 0 {
		title = string(matches[1])
	}
	site.tite = title
```

**Comment by sharadbhat** [comment]
Won't using only the title tag give rise to conflicts? If the title tag does not change based on the page. Title + some UID might be a better approach I think.

**Comment by lucasturci** [comment]
Hey, so, after I implemented this, I realized there may be two types of issues: encoding and titles that [have '/'](https://stackoverflow.com/questions/9847288/is-it-possible-to-use-in-a-filename). I don't use windows, but there may be some issues with '\' too. How should I escape this characters? I think we will have to choose a different sign to replace them, because there is no way to escape them, I guess... To know better what I'm talking about, clone my [fork](https://github.com/lucasturci/stashbox)  and try to run `go run cmd/stashbox/main.go  -url https://github.com/zpeters/stashbox`

The encoding issue is that some sites that are not encoded with utf8 will have bytes not understood by the filesystem, but I don't know if this PR should tackle this problem or another one.
